### ISSUE TITLE
about: "VM Exception while processing transaction: out of gas" on ganache

### ISSUE THREAD

**Issue by crackwei** [zeppelinos/zos#81]
hello,i use truffle and ganache made a demo,but after i update,it show to me the error:"VM Exception while processing transaction: out of gas",how to fits this problem.this is my step:
1. cd myapp
2. truffle unbox webpack
3.npm init
4.zos init myapp
5.in myapp path:npm install zos-lib
6.edit MetaCoin.sol:

```solidity
import "./ConvertLib.sol";
import "zos-lib/contracts/migrations/Migratable.sol";
// This is just a simple example of a coin-like contract.
// It is not standards compatible and cannot be expected to talk to other
// coin/token contracts. If you want to create a standards-compliant
// token, see: https://github.com/ConsenSys/Tokens. Cheers!

contract MetaCoin is Migratable{
	mapping (address => uint) balances;

	event Transfer(address indexed _from, address indexed _to, uint256 _value);

 function initialize(uint _coin, uint x) isInitializer("MetaCoin", "0") public {
		balances[tx.origin] = _coin * x;
	}

	function sendCoin(address receiver, uint amount) public returns(bool sufficient) {
		if (balances[msg.sender] < amount) return false;
		balances[msg.sender] -= amount;
		balances[receiver] += amount;
		emit Transfer(msg.sender, receiver, amount);
		return true;
	}

	function getBalanceInEth(address addr) public view returns(uint){
		return ConvertLib.convert(getBalance(addr),2);
	}

	function getBalance(address addr) public view returns(uint) {
		return balances[addr];
	}
  function test() public returns(uint){
   return 1;
  }
}
```

7. zos add MetaCoin
8. truffle compile
9.zos push --network ganache
  for this step,it shown: **MetaCoin deployment failed with: Error: VM Exception while processing transaction: out of gas** but if i run "truffle migrate --network ganache" first,"zos push" is OK!!!

10.zos create MetaCoin --init initialize --args 10,10 --network ganache

for now ,it run ok

11. update MetaCoin.sol:

```solidity
function test() public returns(uint){
   return 2;
  }
```

12.zos push --network ganache

it will show  **MetaCoin deployment failed with: Error: VM Exception while processing transaction: out of gas**  again,and cannot to resolve,what can i do?

the follow are my truffle and ganache version:

root@ReborTrack:/home/eth/truffle_test# ganache-cli --version
**Ganache CLI v6.1.8 (ganache-core: 2.2.1)**
root@ReborTrack:/home/eth/truffle_test# truffle version
**Truffle v4.1.14 (core: 4.1.14)**
**Solidity v0.4.24 (solc-js)**

**Comment by crackwei** [comment]
root@ReborTrack:/home/eth/truffle_test# ganache-cli --version
**Ganache CLI v6.1.8 (ganache-core: 2.2.1)**
root@ReborTrack:/home/eth/truffle_test# truffle version
Truffle v4.1.14 (core: 4.1.14)
**Solidity v0.4.24 (solc-js)**

**Comment by spalladino** [comment]
@crackwei could you provide your `truffle.js` as well?

**Comment by spalladino** [comment]
@crackwei can you try removing the function `getBalanceInEth`? I'm afraid this could be caused by the zOS CLI not linking the library before deploying.

**Comment by crackwei** [comment]
great,it work good now.thank you very much.
another question,if i keep `getBalanceInEth` ,i must "zos add ConvertLib" first,right?

**Comment by spalladino** [comment]
I've just logged issue #85 to fix this. For now, my suggestion would be to copy the `convert` method of `ConvertLib` into your `MetaCoin` contract, until we can ship a fix for this problem. Apologies for the inconvenience!

**Comment by facuspagnuolo** [comment]
Considering fixed based on #85
